src/field.js

```diff
@@ -15,15 +13,7 @@ class Field {
       return this.options.default
     }
 
-    const type = this.type
-    if (type === Number) return 0
-    if (type === String) return ""
-    if (type === Date) return null
-    if (type === Boolean) return false
-    if (Array.isArray(type)) return []
-    if (type.prototype instanceof BaseEntity) return new type()
-
-    return undefined
+    return null
```

Comment:
Because `null` represents a type of existing property but has no value, instead of undefined whose represents a not defined and has never been initialized.

Reply:
exactly. since in JS we can define a uninitialized variable with `undefined`, and `null` been the explicitly emptiness, gotu could initialize the properties with `undefined` and reserve `null` for when developers explicitly want to make a property empty. [1]

[1] https://vvkchandra.medium.com/essential-javascript-mastering-null-vs-undefined-66f62c65d16b

Reply:
For me the `undefined `makes more sense than the `null`.

undefined: primitive value used when a variable has not been assigned a value.
null: a primitive value to represent an “intentional” absence of any other assigned value.